DigitalOffice/Services/AuthenticationService/Commands/UserLoginCommand.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Text;
+using FluentValidation;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Cryptography;
+using LT.DigitalOffice.AuthenticationService.Token;
+using LT.DigitalOffice.AuthenticationService.Models;
+using LT.DigitalOffice.AuthenticationService.Commands.Interfaces;
+using LT.DigitalOffice.AuthenticationService.Broker.Senders.Interface;
+
+namespace LT.DigitalOffice.AuthenticationService.Commands
+{
+    public class UserLoginCommand : IUserLoginCommand
+    {
+        private readonly NewToken token;
+        private readonly ISender<string, object> authSender;
+        private readonly IValidator<UserLoginCredentialsRequest> validator;
+
+        public UserLoginCommand(
+            [FromServices] NewToken token,
+            [FromServices] IValidator<UserLoginCredentialsRequest> validator,
+            [FromServices] ISender<string, object> authSender)
+        {
+            this.token = token;
+            this.validator = validator;
+            this.authSender = authSender;
+        }
+
+        public async Task<UserLoginResult> Execute(UserLoginCredentialsRequest userCredentials)
+        {
+            validator.ValidateAndThrow(userCredentials);
+
+            object brokerResponse = await authSender.GetResponseFromBroker(userCredentials.Email);
+
+            var brokerResponseCast = Cast(brokerResponse, new { UserId = new Guid(), PasswordHash = ""});
```

Comment:
If the sender returns IUserCredentialsResponse, no cast is needed

Reply:
Fixed